arch/arm/src/common/arm_assert.c

```diff
@@ -411,6 +425,11 @@ static void arm_dumpstate(void)
   /* Dump the user stack */
 
   arm_dump_stack("User", sp,
+#ifdef CONFIG_STACK_COLORATION
+                 up_check_stack_remain(),
```

Comment:
let's rename do_stackcheck to arm_check_stack and call this function inside arm_dump_stack?

Reply:
Done! patch 2